lib/puppet/provider/package/portage.rb

```diff
@@ -111,7 +111,14 @@ def qatom
     @atom ||= begin
       package_info = {}
       # do the search
-      search_output = qatom_bin(*([@resource[:name], '--format', output_format]))
+      should = @resource[:ensure]
+      case should
+      when true, false, Symbol
```

Comment:
I *think* what this is doing is relying on `present`, `absent`, and `latest` becoming symbols, while versions are going to come through as strings?

Could you add a comment describing that so someone not super familiar with Puppet and Ruby can understand what's going on here? It's a big magical.

Reply:
It also may be clearer to invert the logic here - `when String` for when we have a version, and the else block can handle everything else

Reply:
Hi, i will complete these tasks during next week.

Reply:
Hi, i've put a comment in front of this code. But i would keep the logic here consistent with the other package providers like apt and yum, where we use the same logic.